src/service_control_client_impl.cc

```diff
@@ -265,22 +265,29 @@ void ServiceControlClientImpl::Quota(
                 quota_request);
 
     std::shared_ptr<QuotaAggregator> quota_aggregator_copy = quota_aggregator_;
-    quota_transport(*quota_request_copy, quota_response,
-                    [this, quota_aggregator_copy, quota_request_copy,
-                     quota_response, on_quota_done](Status status) {
-
-                      if (status.ok()) {
-                        quota_aggregator_copy->CacheResponse(
-                            *quota_request_copy, *quota_response);
-                      } else {
-                        GOOGLE_LOG(ERROR) << "Failed in Check call: "
-                                          << status.error_message();
-                      }
-
-                      delete quota_request_copy;
-
-                      on_quota_done(status);
-                    });
+    quota_transport(*quota_request_copy, quota_response, [this,
+                                                          quota_aggregator_copy,
+                                                          quota_request_copy,
+                                                          quota_response,
+                                                          on_quota_done](
+                                                             Status status) {
+
+      if (status.ok()) {
+        quota_aggregator_copy->CacheResponse(*quota_request_copy,
+                                             *quota_response);
+      } else {
+        // on network error, failed open, reset in_flight flag to false
+        ::google::api::servicecontrol::v1::AllocateQuotaResponse dummy_response;
```

Comment:
You can remove these name space prefix, there is a "using" in line 27.  please remove it in other places too, such as line 262